### PR TITLE
Fixes #34378 - Use process abstractions from sp-dynflow

### PR DESCRIPTION
**Pull request by adamruzicka**
Requires:
- https://github.com/theforeman/smart_proxy_dynflow/pull/103

TODO:
- [ ] bump sp-dynflow dependency once it gets released

### CODE REVIEW

**Comment by ezr-ondrej** [comment]
This has a conflict now, but I believe this is the next one on the list? I'll take it for a spin, can you rebase? :)

File: lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb [hunk 1]
```diff
@@ -387,9 +355,9 @@ def upload_file(local_path, remote_path)
     end
 
     def ensure_remote_directory(path)
-      exit_code, _output, err = run_sync("mkdir -p #{path}")
-      if exit_code != 0
-        raise "Unable to create directory on remote system #{path}: exit code: #{exit_code}\n #{err}"
+      pm = run_sync("mkdir -p #{path}")
+      if pm.status != 0
+        raise "Unable to create directory on remote system #{path}: exit code: #{pm.status}\n #{pm.stderr.to_s.chomp}"
```
Comment (ezr-ondrej) [inline lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb:360]: I've had the host down and this message appeared and I found it bit confusing, can we have something bit more telling?

To be frank next line was `ssh: connect to host gina-langhorn.oezr.local port 22: No route to host` so it is ok, it is just bit harder to read.

**Review by ezr-ondrej — COMMENTED** [review]
Just one comment (just UX), in general I see it working as expected! :+1:

File: lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb (hunk 1 above)
Comment (adamruzicka) [reply lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb:360]: I'm trying to address this in the followup pr https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/72

**Review by ezr-ondrej — APPROVED** [review]
Thanks @adamruzicka !

**Pull request merged**